### ISSUE TITLE
Continous loop due to class instance

### ISSUE THREAD

**Issue by Julian-Theis** [spark-jobserver/spark-jobserver#643]
Hi,

my JavaSparkJob runs into a continous loop as soon as I am creating a new instance of a class which processes JavaRDD's.
My current code looks like the following. It's working fine on Spark natively without the JobServer. Do I miss anything?

```
public class SomeJob extends JavaSparkJob{

public Object runJob(JavaSparkContext jsc, Config jobConfig) {
		JavaRDD<String> someRDD = ... // Processing - works fine!
                
               TestClass tcl = new TestClass();
               tcl.setRDD(someRDD);
               tcl.getRDD();
      }
}
```

```
public class TestClass implements Serializable{

	private JavaRDD<String> rdd;
	
	public TestClass(){
	}

	public void setRDD(JavaRDD<String> someRDD){
		rdd = someRDD;
	}
	
	public JavaRDD<String> getRDD(){
		return rdd;
	}
}
```

**Comment by hntd187** [comment]
I'm not sure what you're asking and there are no loops in this Job, so I don't know how an infinite loop would happen if that's what you are referring to.

**Comment by Julian-Theis** [comment]
Well, it is pretty weird:
I am running the example described above without errors, but the Job is never ending (at least using my Spark configuration on Mesos) ?! I've terminated it after a couple of hours without any suspicious log entry. In contrast, as soon as I am moving the JavaRDD processing functions from an "external" class instance directly to the runJob function, it is working fine. 
As I got a little bit annoyed, I tried using the Scala API instead. Lo and behold, initializing the job as well as a class instance processing JavaRDDs works as expected. So I really don't know why the example using the Java API fails.

I'll continue keeping track of it and will provide some more information as soon as  I am observing further conspicuities...
